02 fridge fleet monitor (#174)
blues/app-accelerators: 02-fridge-fleet-monitor/README.md

![Diagram of Breadboard Circuit](images/fridge-fleet-monitor-breadboard.png)

_Breadboard Connections_

## Hardware Instalation

With the hardware assembled and the proper code on the Pico, attach the temperature sensor inside the fridge you wish to monitor.  Be careful to route the wires so they do not affect the seal of the door.

To attach the sensor inside the fridge, use Command Strips or some other adhesive strip attached to the back of the sensor, and wipe down the area to place with node with rubbing alcohol prior to attaching for best adhesion.

![DHT11 Sensor on side wall](images/dht11-mounted-fridge.jpg)

_DHT11 Sensor installed in a refrigeration unit._

Likewise, to install the door sensors, attach each piece of the sensor on one side of the door via Command Strips and ensure they line up as closely as possible. See the image below.

![door switches aligned together when door is closed](images/door-switches-closed.jpg)

## Testing

Ensure your Notecarrier is powered with a battery or via a USB power supply.

Connect the Raspberry Pi Pico to your computer with a USB Cable, ensuring you have no power cable between the Notecarrier and the Pico to prevent damage.

> Open a Serial Terminal Emulator to view the log output from the Pico Code

Reviewer: ```suggestion
Open a Serial Terminal Emulator to view the log output from the Pico code.
```

You should see output similar to below:

![example output from serial console of Pico](images/pico-serial-output.png)

> Open the fridge door.  In the console you should see the message 'Door Wakeup. Door Value: True' and 'Door Open'.  In addition, you should see an event in the Blues Notehub with a filename of `door.qo` with the body of `{'open':true}`

Reviewer: ```suggestion
Open the fridge door.  In the console you should see the message 'Door Wakeup. Door Value: True' and 'Door Open'.  In addition, you should see an event in the Blues Notehub with a filename of `door.qo` with the body of `{"open":true}`
```

> Close the fridge door. In the console you should see the message 'Door Closed'.  In addition, you should see an event in the Blues Notehub with a filename of `door.qo` with the body of `{'closed':true}`

Reviewer: ```suggestion
Close the fridge door. In the console you should see the message 'Door Closed'.  In addition, you should see an event in the Blues Notehub with a filename of `door.qo` with the body of `{"closed":true}`
```

> Aditionally you should also see an event in Notehub with the filename of `readings.qo` and the body similar to `{"humidity":48,"temperature":22}`. This is the relative humidity percentage and the temperature in celcius.  This event should be sent by default every 60 minutes.

Reviewer: ```suggestion
Additionally you should see an event in Notehub with the filename of `readings.qo` and a body similar to `{"humidity":48,"temperature":22}`. This is the relative humidity percentage and the temperature in Celsius.  This event should be sent by default every 60 minutes.
```

## Further Development

If you wish to develop this project further, here are some suggestions.

> Firstly, the DHT11 sensor is cheap and accessable, but not nescarcellary the most accurate.  Consider replacing it with another, for example the BME280 which connects via I2C, so will share the same connections as the Notecarrier, and there is a [CircutPython Library available as well](https://docs.circuitpython.org/projects/bme280/en/latest/)

Reviewer: ```suggestion
Firstly, the DHT11 sensor is cheap and accessible, but not necessarily the most accurate.  Consider replacing it the BME280, which also connects via I2C, so it will share the same connections as the Notecarrier. There is a [CircutPython Library available for the BME280 as well](https://docs.circuitpython.org/projects/bme280/en/latest/)
```

> Secondly, a great addition to the functionality would be to add an note to be sent if the fridge door is left open for a configurable time.  I will leave this as an exercise to the user on how to add this feature, but it should be quite simple with the existing code.

Reviewer: ```suggestion
Secondly, a great addition to the functionality would be to add and send Note if the fridge door is left open for a configurable time.  I will leave this as an exercise to the user on how to add this feature, but it should be quite simple with the existing code.
```
